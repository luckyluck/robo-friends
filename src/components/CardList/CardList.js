import React from 'react';

import Card from '../Card/Card';

import './CardList.css';

export const CardList = ({ robots, isPending }) => {
    if (isPending) {
        return (
            <h1 className="f1 tc">Loading...</h1>
        );
    } else {
        return (
            <div className="CardList">
                {robots.map(
                    robot => <Card
                        key={robot.id}
                        id={robot.id}
                        name={robot.name}
                        email={robot.email}
                    />
                )}
            </div>
        );
    }
};

export default CardList;