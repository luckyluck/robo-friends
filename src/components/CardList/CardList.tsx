import * as React from 'react';

import Card from '../Card/Card';

import './CardList.css';

interface ICardListProps {
    robots: Array<{
        id: string,
        name: string,
        email: string
    }>,
    isPending: boolean
}

export const CardList = ({ robots, isPending }: ICardListProps) => {
    if (isPending) {
        return (
            <h1 className="f1 tc">Loading...</h1>
        );
    } else {
        return (
            <div className="CardList">
                {robots.map(
                    robot => <Card
                        key={robot.id}
                        id={robot.id}
                        name={robot.name}
                        email={robot.email}
                    />
                )}
            </div>
        );
    }
};

export default CardList;