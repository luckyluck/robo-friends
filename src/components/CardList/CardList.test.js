import React from 'react';
import { shallow } from 'enzyme';

import { CardList } from './CardList';

describe('CardList component', () => {
    it('should shallowly render component', () => {
        const mockRobots = [
            {
                id: 1,
                name: 'John Snow',
                username: 'JohnJohn',
                email: 'john@gmail.com'
            }
        ];
        expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
    });
});