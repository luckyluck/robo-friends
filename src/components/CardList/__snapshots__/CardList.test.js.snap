// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component should shallowly render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    robots={
        Array [
            Object {
              "email": "john@gmail.com",
              "id": 1,
              "name": "John Snow",
              "username": "JohnJohn",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="john@gmail.com"
          id={1}
          name="John Snow"
/>,
      ],
      "className": "CardList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "john@gmail.com",
          "id": 1,
          "name": "John Snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="john@gmail.com"
            id={1}
            name="John Snow"
/>,
        ],
        "className": "CardList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "john@gmail.com",
            "id": 1,
            "name": "John Snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
